User_Story#1/SortingApp/SortinfApp.UnitTests/Utilities/Validator/LocalFileValidatorTest.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.IO;
+using BusinessLayer.Validator;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SortinfApp.UnitTests.Utilities.Validator
+{
+    /// <summary>
+    /// Represents tests for LocalFileValidator class. 
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class LocalFileValidatorTest
+    {
+        /// <summary>
+        /// Tests IsDataExist if pass empty path it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void LocalFileValidator_IsDataExist_PassEmtptyPath_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var localFileValidator = new LocalFileValidator();
+            string path = null;
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => localFileValidator.IsDataExist(path));
+        }
+
+        /// <summary>
+        /// Tests IsDataExist if pass exist file it will get true.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void LocalFileValidator_IsDataExist_PassExistFile_GetTrue()
+        {
+            //
+            // Arrange.
+            //
+            var localFileValidator = new LocalFileValidator();
+            string path = @"D:\collectionToRead.txt";
+            bool actualResult;
+
+            //
+            // Act.
+            //
+            using (File.Create(path)) { }
```

Comment:
We need to check if path is valid before we creating the file.
Directory.Exists(path)

Reply:
Do you mean that such path could be not valid for some user and I should prevent creating file in not valid place? But in this case result of test case would be not pass too as when path not valid.

Reply:
Lets imagine a test-method that checks if a data exists.
In the method we specified a path that is not valid.
In that case method will work.
But we will get an answer about 'Does logic work correct if the path is incorrect'
And we will not get an answer about 'Does logic work correct when the data exists'.

So these are two different test-cases that need to be checked separately.